testplan/web_ui/testing/src/Report/reportUtils.js

```diff
@@ -102,8 +102,8 @@ const propagateIndicesRecur = (entries, parentIndices) => {
       caseCount.failed += descendantsIndices.case_count.failed;
     } else {
       // Count testcase's status.
-      caseCount.passed += entry.status === 'passed';
-      caseCount.failed += entry.status === 'failed';
+      caseCount.passed += ['passed', 'xpass'].includes(entry.status);
```

Comment:
This is fine for now but in future I think we should be keeping these pass/fail counts in the backend and not calculating them here.